tests/test_conn.py

```diff
@@ -116,3 +116,125 @@ def test_delete_user_mailcow_missing(conn, mailcow):
 def test_db_version(conn):
     version = conn.get_dbversion()
     assert type(version) == int
+
+
+def test_users_to_warn(conn, monkeypatch):
+    yeartoken = conn.add_token("1y", token="asd1", expiry="1y", prefix="tmp.")
+    monthtoken = conn.add_token("1m", token="asd2", expiry="1m", prefix="tmp.")
+    weektoken = conn.add_token("1w", token="asd3", expiry="1w", prefix="tmp.")
+    daytoken = conn.add_token("1d", token="asd4", expiry="1d", prefix="tmp.")
+
+    yearuser = conn.add_email_account(yeartoken)
+    monthuser = conn.add_email_account(monthtoken)
+    weekuser = conn.add_email_account(weektoken)
+    dayuser = conn.add_email_account(daytoken)
+
+    sysdate = int(time.time())
+    # add 20 hours to time
+    sysdate += 60 * 60 * 20
+    users = conn.get_users_to_warn(sysdate)
+    assert len(users) == 1
+    assert "360 minutes" in users[0]["message"]
+    assert users[0]["user"].addr == dayuser.addr
+    # don't remember who was warned already
```

Comment:
this is rather a comment on get_users_to_warn invocation further up, right?

Reply:
no, because `user_was_warned` is the function which *does* the remembering. I renamed it to `remember_warning` now.